java/ql/test/library-tests/dataflow/range-analysis-inline/B.java

```diff
@@ -106,15 +106,53 @@ static void arraylength(int[] arr) {
         }
     }
 
-    static int varbound(int b) {
+    static int varBound(int b) {
         bound(b);
         int result = 0;
         for (int i = 0;
             i < b;
             i++) { // $ bound="i <= b - 1"
             result = i; // $ bound="i <= b - 1"
         }
+        return result; // We cannot conclude anything here, since we do not know that b > 0
+    }
+
+    static int varBoundPositiveGuard(int b) {
+        bound(b);
+        if (b > 0) {
+            int result = 0;
+            for (int i = 0;
+                i < b;
+                i++) { // $ bound="i <= b - 1"
+                result = i; // $ bound="i <= b - 1"
+            }
+            return result; // $ MISSING: bound="result <= b - 1"
```

Comment:
Btw. this is actually a case that potentially could be improved by some sort of "loop executes at least once" analysis.